### PR TITLE
feat: fee modal for BTC

### PR DESCRIPTION
**Pull request by edgarkhanzadian**
Implemented fee modal for BTC (STX is on the way)

https://github.com/user-attachments/assets/1b9289ad-09d5-4859-8e26-3e1be8dfd077

### CODE REVIEW

File: apps/mobile/src/features/approver/components/bitcoin-fee-card.tsx [hunk 1]
```diff
@@ -48,27 +30,16 @@ export function BitcoinFeeCard({ feeType, amount }: FeeCardProps) {
...
         <FeeBadge type="normal" />
       </Box>
-
       <Box mx="-5">
-        <Cell.Root pressable={true} onPress={() => {}}>
+        <Cell.Root pressable onPress={onPress}>
           <Cell.Icon>
             <Avatar bg="ink.background-secondary" p="2">
-              {feeIcon}
+              {getFeeData(feeType).icon}
```
Comment (pete-watters) [inline apps/mobile/src/features/approver/components/bitcoin-fee-card.tsx:37]: Can you just call `getFeeData(feeType)` once instead? 

```
const { icon, title, time } = getFeeData(feeType);
```

File: apps/mobile/src/features/psbt-signer/psbt-signer.tsx [hunk 2]
```diff
@@ -1,41 +1,114 @@
...
   onEdit(): void;
   onSuccess(): void;
 }
 
-export function PsbtSigner({ psbtHex, onEdit, onSuccess }: PsbtSignerProps) {
-  const network = useCurrentNetworkState();
+function getFeeType({
+  psbtFee,
+  fees,
+}: {
```
Comment (pete-watters) [inline apps/mobile/src/features/psbt-signer/psbt-signer.tsx:61]: Can you move this type out to an `interface`?

File: apps/mobile/src/features/psbt-signer/psbt-signer.tsx [hunk 3]
```diff
@@ -71,71 +198,88 @@ export function PsbtSigner({ psbtHex, onEdit, onSuccess }: PsbtSignerProps) {
...
-          titleOpened={t({
-            id: 'approver.advanced.hide',
-            message: 'Hide advanced options',
-          })}
-        >
-          <Approver.Section noTopPadding>
-            <InputsAndOutputsCard
-              inputs={psbtDetails.psbtInputs}
-              outputs={psbtDetails.psbtOutputs}
+    <>
```
Comment (pete-watters) [inline apps/mobile/src/features/psbt-signer/psbt-signer.tsx:201]: Maybe you can move this to a `.layout` file?

File: apps/mobile/src/features/psbt-signer/psbt-signer.tsx [hunk 4]
```diff
@@ -71,71 +198,88 @@ export function PsbtSigner({ psbtHex, onEdit, onSuccess }: PsbtSignerProps) {
...
+      <Approver requester="https://leather.io">
```
Comment (pete-watters) [inline apps/mobile/src/features/psbt-signer/psbt-signer.tsx:202]: I know it was already like this but should `requester` be hardcoded?

File: apps/mobile/src/features/approver/utils.tsx [hunk 5]
```diff
@@ -0,0 +1,72 @@
+import { t } from '@lingui/macro';
```
Comment (pete-watters) [inline apps/mobile/src/features/approver/utils.tsx:1]: Maybe you can just call this file `get-fee-data.ts`?

File: apps/mobile/src/features/approver/components/fees-sheet.tsx [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+  function getUsd(fee: number) {
+    if (!fee) return createMoney(0, 'USD');
+    const btcBalance = createMoney(fee, 'BTC');
+    const usdBalance = baseCurrencyAmountInQuoteWithFallback(btcBalance, btcMarketData);
+    return usdBalance;
+  }
+
+  function getFee(feeType: FeeTypes) {
+    const feeRate = match<FeeTypes>()(feeType, {
+      [FeeTypes.Low]: fees?.hourFee ?? BigNumber(0),
```
Comment (pete-watters) [inline apps/mobile/src/features/approver/components/fees-sheet.tsx:50]: Maybe we can just call `BigNumber(0)` once?

`const bigZero = BigNumber(0);`

**Comment by pete-watters** [comment]
Nice work @edgarkhanzadian 👍 I left a few comments, mostly minor things. 

I tried it out and its working quite well but I have a couple of questions:
1. `custom` fee is disabled but it doesn't look any different than the others. 
    - Maybe we should just hide it instead as it seems broken?
    - we probably need to add a `disabled` style for `Pressable`
2. maybe a seperate bug but on the `Send` form 
    - the `Review` button doesn't appear active if you enter recipient and then amount
    - you can still click `Review` but the button is greyed out
    
Screenshot 2024-12-20 at 12 00 45

3. if you try and send funds from an empty account it doesn't show any errors
4. the drawers for recipient + fee stay open even after you select / populate so you need to tap again to close them

**Review by fbwoolf — APPROVED** [review]

**Pull request merged**